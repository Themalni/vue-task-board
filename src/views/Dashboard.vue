<template>
  <Board :header="boardHeader">
    <Board-Column v-for="category in categories" :key="category.id" :category="category"/>
  </Board>
</template>

<script>
import Board from '@/components/Board'
import BoardColumn from '@/components/BoardColumn'
import { mapGetters } from 'vuex'

export default {
  name: 'Dashboard',
  components: {
    Board,
    BoardColumn
  },
  data: () => ({
    boardHeader: 'Tablica'
  }),
  computed: {
    ...mapGetters(['categories'])
  }
}
</script>
