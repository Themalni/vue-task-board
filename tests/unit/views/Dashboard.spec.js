import { mount, createLocalVue } from '@vue/test-utils'
import Vuex from 'vuex'
import mockState from '../utils/mockState'
import Dashboard from '@/views/Dashboard'

const localVue = createLocalVue()
localVue.use(Vuex)

describe('Dashboard', () => {
  let wrapper
  let store
  let state
  let getters

  beforeEach(() => {
    state = mockState
    getters = {
      categories: () => state.categories,
      newModalIsOpen: () => false
    }
    store = new Vuex.Store({
      state,
      getters
    })

    wrapper = mount(Dashboard, {
      data () {
        return {
          boardHeader: 'Tablica'
        }
      },
      store,
      localVue,
      sync: false
    })
  })

  afterEach(() => {
    wrapper.destroy()
  })

  it('renders correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('does contain `Board` element', () => {
    expect(wrapper.classes('board')).toBe(true)
  })
})
