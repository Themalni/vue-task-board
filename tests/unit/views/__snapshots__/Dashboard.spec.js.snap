// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders correctly 1`] = `
<div class="board">
  <div class="board-header">
    <h1 class="board-header-text">Tablica</h1>
    <div class="mb-2"><button class="button button-success">Dodaj zadanie</button></div>
  </div>
  <div class="board-body"></div>
  <!---->
</div>
`;
